omnigan/trainer.py

```diff
@@ -940,6 +974,30 @@ def eval(self, num_threads=5, verbose=0):
                         self.logger.losses.task_loss[update_task][
                             domain
                         ] = update_loss.item()
+
+                    if update_task == "m":
+                        # ? output features classifier
+                        prediction = self.G.decoders[update_task](self.z)
+                        task_tensors[update_task] = prediction
+
+                        # Main loss first:
+                        update_loss = (
+                            self.losses["G"]["tasks"][update_task]["main"](
+                                prediction, update_target
+                            )
+                            * lambdas.G[update_task]["main"]
+                        )
+                        self.logger.losses.task_loss[update_task]["main"][
+                            domain
+                        ] = update_loss.item()
+
+                        # Then TV loss
+                        update_loss = self.losses["G"]["tasks"][update_task]["tv"](
+                            prediction
+                        )
+                        self.logger.losses.task_loss[update_task]["tv"][
+                            domain
+                        ] = update_loss.item()
```

Comment:
@vict0rsch this is the part of the code I'm worried about. This could get messy if we have to treat each auxiliary task independently like this. Though it might be the case that this is only necessary for the mask decoder since that's our focal task and the other auxiliary tasks may not require additional losses.